### PR TITLE
Broadcast source unit test inval id base

### LINKED ISSUE #57457: LE Audio: BAP Broadcast Source Unit test
**Is your enhancement proposal related to a problem? Please describe.**
This is mainly a training exercise for @Thalley to get experience with the unit test framework(s) using a fairly simply module (the BAP Broadcast Source). 

**Describe the solution you'd like**
Implement unit tests for the BAP Broadcast Source to get at least 80% test coverage using only those tests. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

```[tasklist]
### Tasks
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62162
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62257
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62259
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62338
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62344
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62348
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62361
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62402
- [ ] https://github.com/zephyrproject-rtos/zephyr/pull/62437
```

### PR DESCRIPTION
**Pull request by Thalley**
fixes https://github.com/zephyrproject-rtos/zephyr/issues/57457

### CODE REVIEW

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 1]
```diff
@@ -1283,3 +1345,110 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base)
...
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(fixture->source, &base_buf);
+	zassert_not_equal(0, err, "Did not fail with too small base_buf (%u)", base_buf.size);
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
+}
```
Comment (MariuszSkamra) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:1454]: It looks like those two are testing exactly the same. Are you sure we need separate tests for `very_small_buf` and `small_buf`?
I do not get which buffer size values can be considered as very small, and which are small ^^

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 2]
```diff
@@ -1283,3 +1345,110 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base)
...
+	int err;
+
+	printk("Creating broadcast source with %zu subgroups with %zu streams\n",
+	       create_param->params_count, fixture->stream_cnt);
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(fixture->source, NULL);
+	zassert_not_equal(0, err, "Did not fail with null ID");
```
Comment (MariuszSkamra) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:1382]:
```suggestion
	zassert_not_equal(0, err, "Did not fail with null buffer pointer");
```

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 3]
```diff
@@ -1164,6 +1164,68 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_id)
...
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_id(NULL, &broadcast_id);
+	zassert_not_equal(0, err, "Did not fail with null source");
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
```
Comment (MariuszSkamra) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:1184]: Is there a need to create a source to call `bt_bap_broadcast_source_get_id` with null source?

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c [hunk 4]
```diff
@@ -1283,3 +1345,110 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base)
...
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(NULL, &base_buf);
+	zassert_not_equal(0, err, "Did not fail with null source");
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
```
Comment (MariuszSkamra) [inline tests/bluetooth/audio/bap_broadcast_source/src/main.c:1367]: ditto, source does not seem to be used in this test scenario

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c (hunk 3 above)
Comment (Thalley) [reply tests/bluetooth/audio/bap_broadcast_source/src/main.c:1184]: Nope :) Removed

File: tests/bluetooth/audio/bap_broadcast_source/src/main.c (hunk 1 above)
Comment (Thalley) [reply tests/bluetooth/audio/bap_broadcast_source/src/main.c:1454]: Right, so the "very small buf" is a buffer that is smaller than `MINIMUM_BASE_SIZE` and fails as part of 
```c
	if ((buf->size - buf->len) < MINIMUM_BASE_SIZE) {
		return false;
	}
```
in `encode_base`.

The "small buf" is a test that verifies that the buffer supplied can hold a minimum BASE, but not the BASE we supply, and will fail in one of the dynamic checks like
```c
	if ((buf->size - buf->len) < sizeof(len)) {
		LOG_DBG("No room for metadata length");

		return false;
	}
```

So the two tests different error conditions (and thus more code coverage), and therefore I believe both should exist.

**Comment by MariuszSkamra** [comment]
@Thalley Are you about to update this PR? I think you forgot to push, looking at your comments ^^

**Comment by Thalley** [comment]
> @Thalley Are you about to update this PR? I think you forgot to push, looking at your comments ^^

Thanks - Must've pushed some other branch :D

**Review by MariuszSkamra — APPROVED** [review]

**Review by fredrikdanebjer — APPROVED** [review]

**Pull request merged**